src/app/components/charts/chart.service.js

```diff
@@ -183,19 +201,26 @@ angular.module('transmartBaseUi')
              * @private
              */
             var _addLabel = function (obs, node) {
-                var present = false;
-                cs.labels.forEach(function(label){
-                    present = label.label === obs.label ? true : present;
-                });
-                if(!present){
-                    cs.labels.push({
+                var label = _.findWhere(cs.labels, {label: obs.label});
+                if(!label){
+                  if(cs.numDim < cs.maxDim)
+                  {
+                    cs.numDim++;
+                    label = {
                         label: obs.label,
                         type: typeof obs.value,
                         name: _getLastToken(obs.label),
                         ids: cs.chartId++,
-                        study: node.study
-                    });
+                        study: node.study,
+                        rendered: false
+                    };
+                    cs.labels.push(label);
+                  } else {
+                    AlertService.add('danger', 'Max number of dimensions reached !', 2000);
+                  }
                 }
+                if(label && label.type === 'number' && (obs.value % 1) != 0)
```

Comment:
I agree (obs.value % 1) != 0) is better for some cases but it cant check for the actual presicion.
With  (min + '').split('.') we can do:
precision = (min + '').split('.');
numbersAfterTheDot = precision[1].length